academic_validation_framework/credibility/base.py

```
@@ -0,0 +1,408 @@
+    def _calculate_composite_score(
+        self, 
+        scores: Dict[str, float], 
+        weights: Optional[Dict[str, float]] = None
+    ) -> float:
+        """Calculate a weighted composite score."""
+        if not scores:
+            return 0.0
+        
+        if weights is None:
+            # Equal weights if not specified
+            weights = {key: 1.0 for key in scores.keys()}
```

Ruff (0.11.9)

Use `key in dict` instead of `key in dict.keys()`

Remove `.keys()`

(SIM118)